src/display/text_layer.js

```diff
@@ -528,20 +534,48 @@ var renderTextLayer = (function renderTextLayerClosure() {
     },
 
     _render: function TextLayer_render(timeout) {
-      var textItems = this._textContent.items;
-      var textStyles = this._textContent.styles;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(this, textItems[i], textStyles);
+      let capability = createPromiseCapability();
+      let pump;
+      if (this._textContent === undefined) {
+        pump = () => {
+          reader.read().then(({ value, done, }) => {
+            if (done) {
+              capability.resolve();
+              return;
+            }
+            Util.appendToArray(this._textContent.items, value.items);
```

Comment:
> We can process items as they coming in.

Not sure if i am getting you properly here. Do we need to have all styles cached before we start processing items, or we can build stylesCache on the fly during processing of items, something like:

```javascript
pump = () => {
  this._reader.read().then(({ value, done, }) => {
  if (done) {
    return;
  }

  Util.extendObj(styleCache, value.styles);
  this._processItems(value.items, stylesCache);
  pump();

  }, () => {
    this._capability.reject('Error in reading chunks');
  });
};
```

> Create e.g. `processItems(items, stylesCache)` and use it for a chunk

Can you please elaborate a bit `processItems(items, stylesCache)` working.